packages/plugin-pt15/src/algorithm.ts

```diff
@@ -0,0 +1,214 @@
+import { AnyIndexStore, AnyOrama, SearchableType, Tokenizer, VectorIndex } from "@orama/orama"
+import { avl, bkd, flat, bool } from '@orama/orama/trees'
+import {
+  getVectorSize, internalDocumentIDStore, isVectorType } from '@orama/orama/components'
+
+type InternalDocumentID = internalDocumentIDStore.InternalDocumentID;
+
+export type TreeType = 'AVL' | 'Radix' | 'Bool' | 'Flat' | 'BKD';
+export type TTree<T = TreeType, N = unknown> = {
+    type: T;
+    node: N;
+    isArray: boolean;
+};
+export type Tree =
+  // We don't store strings inside a tree
+  // | TTree<'Radix', radix.RadixNode>
+  | TTree<'Position', PositionsStorage>
+  | TTree<'AVL', avl.AVLTree<number, InternalDocumentID[]>>
+  | TTree<'Bool', bool.BoolNode>
+  | TTree<'Flat', flat.FlatTree>
+  | TTree<'BKD', bkd.BKDTree>;
+
+const MAX_POSITION = 15
+export type PositionStorage = Record<string, number[]>
+// 15 `PositionStorage`s
+export type PositionsStorage = [
+  PositionStorage,
+  PositionStorage,
+  PositionStorage,
+  PositionStorage,
+  PositionStorage,
+  PositionStorage,
+  PositionStorage,
+  PositionStorage,
+  PositionStorage,
+  PositionStorage,
+  PositionStorage,
+  PositionStorage,
+  PositionStorage,
+  PositionStorage,
+  PositionStorage,
+]
+
+export interface PT15IndexStore extends AnyIndexStore {
+  indexes: Record<string, Tree>
+  vectorIndexes: Record<string, VectorIndex>
+  searchableProperties: string[]
+  searchablePropertiesWithTypes: Record<string, SearchableType>
+}
+
+function create_obj() {
+  // object with empty prototype to cheap objects
+  return Object.create(null)
+}
+
+export function recursiveCreate<T extends AnyOrama>(indexDatastore: PT15IndexStore, schema: T['schema'], prefix: string) {
```

Comment:
Is recursion well supported in that case? Should we opt for an iterative approach for slightly larger datasets (50k elements)?

Reply:
This is not an intensive path: this function is invoked only during the orama creation